### PR TITLE
refactored news_item to remove #published?

### PR DESCRIPTION
**Pull request by itsWill**
Used a lambda in the validation felt that it was more concise and slightly cleaner than using a method. Though a could
``` ruby
def published?
  state.published?
end
```
also be used.

@cawel thanks for the tip!

### CODE REVIEW

**Comment by cawel** [comment]
:+1: for the lambda.

Bonus: you could also turn that class method into a [scope](http://guides.rubyonrails.org/active_record_querying.html#scopes).

**Comment by itsWill** [comment]
:+1: neat idea.

**Comment by cawel** [comment]
Oh yeah. Always nice to see clean code :)

File: app/models/news_item.rb [hunk 1]
```diff
@@ -1,15 +1,7 @@
 class NewsItem < ActiveRecord::Base
   extend Enumerize
   enumerize :state, in: [:draft, :archived, :published]
-  validates :published_at, presence: true, if: :published?
+  scope :published, -> { where(state: :published).order(:published_at) }
+  validates :published_at, presence: true, if: lambda{ state.published? }
```
Comment (nicholasjhenry) [inline app/models/news_item.rb]: @itsWill Thanks for making this change. One suggestion would be to use consistent lambda syntax. In the scope, the "stabby lambda" `->` (preferred), and in the validation the keyword `lambda` is used.

**Pull request merged**